### ISSUE TITLE
updating magisk manager to v7.1.1 says signature mismatch.

### ISSUE THREAD

**Issue by ashjas** [topjohnwu/Magisk#1311]
Cant install update neither through github apk, nor through magisk manager app.
logcat snippet:

> 04-07 07:35:35.925 32135 32135 E id.defcontaine: Not starting debugger since process cannot load the jdwp agent.
> 04-07 07:35:36.217  5293  5293 I Finsky  : [2] tkl.a(6): Verification requested, id = 20
> 04-07 07:35:36.228  5293  5340 I Finsky  : [622] tlj.a(2): Single user settings service is not running, bind it now
> 04-07 07:35:36.235  5293  5293 I Finsky  : [2] tlk.onServiceConnected(1): Single user settings service is connected
> 04-07 07:35:36.246  5293  5340 W Settings: Setting install_non_market_apps has moved from android.provider.Settings.Global to android.provider.Settings.Secure, returning read-only value.
> 04-07 07:35:36.429  5293  5340 I Finsky  : [622] irp.a(3): Task 7 requested foreground
> 04-07 07:35:36.430  5293  5340 I Finsky  : [622] irp.a(8): Not entering foreground
> 04-07 07:35:36.850  5293  5329 I Finsky  : [612] tqo.b(3): Verification id=20 response=SAFE
> 04-07 07:35:36.860  5293  5293 I Finsky  : [2] trh.c(3): Verifying id=20, result=1
> 04-07 07:35:36.870  5293  5293 I Finsky  : [2] trh.b(6): Verification complete: id=20, package_name=com.topjohnwu.magisk
> 04-07 07:35:36.886  2487  2754 W PackageManager: Package com.topjohnwu.magisk signatures do not match previously installed version; ignoring!
> 04-07 07:35:36.893  1966  4917 E         : Couldn't opendir /data/app/vmdl89155869.tmp: No such file or directory
> 04-07 07:35:36.893  1966  4917 E installd: Failed to delete /data/app/vmdl89155869.tmp: No such file or directory
> 04-07 07:35:36.896  2487  4498 I ActivityManager: Killing 30111:com.google.android.apps.turbo/u0a42 (adj 906): empty #17
> 04-07 07:35:36.896  2487  2644 W libprocessgroup: kill(-30111, 9) failed: No such process
> 04-07 07:35:36.900  2487  4498 I ActivityManager: START u0 {dat=file:///data/user_de/0/com.google.android.packageinstaller/no_backup/package360722467771760176.apk flg=0x2000000 cmp=com.google.android.packageinstaller/com.android.packageinstaller.InstallFailed (has extras)} from uid 10046

**Comment by ashjas** [comment]
I debugged further.
I copied the github version, and one installed on my device and unzipped both.
To my surprize, everything else was exactly same when comparing through beyond compare, apart from:
these files in both apks:
CERT.RSA,CERT.SF,MANIFEST.MF
I am attaching only the contents of these files from both the apks, if anyone wants to look further what actually happened.

Also, at this state, i am unable to update any module zips from within magisk manager, it says Failed without much information. Logcat at module install time if it helps to debug:

> 04-07 08:21:55.521   468  4055 D audio_route: Apply path: low-latency-playback speaker
> 04-07 08:21:55.523   468  4055 D audio_hw_primary: select_devices: done
> 04-07 08:21:55.528   468  4055 D msm8916_platform: platform_set_channel_map mixer_ctl_name:Playback Channel Map12
> 04-07 08:21:55.528   468  4055 D msm8916_platform: platform_set_channel_map: set mapping(1 2 0 0 0 0 0 0) for channel:2
> 04-07 08:21:55.533   485   485 E ANDR-PERF: IPerf::tryGetService failed!
> 04-07 08:21:55.533   485   485 E QCOM PowerHAL: Failed to acquire lock.
> 04-07 08:21:55.535   485   485 E ANDR-PERF: IPerf::tryGetService failed!
> 04-07 08:21:55.535   485   485 E QCOM PowerHAL: Failed to acquire lock.
> 04-07 08:21:55.558   468  4055 E audio_hw_extn: audio_extn_perf_lock_release: Perf lock release error 
> 04-07 08:21:55.558   468  4055 D audio_hw_primary: start_output_stream: exit
> 04-07 08:21:55.600   485   485 E ANDR-PERF: IPerf::tryGetService failed!
> 04-07 08:21:55.600   485   485 E QCOM PowerHAL: Failed to acquire lock.
> 04-07 08:21:55.687   511  1190 W SurfaceFlinger: Attempting to set client state on removed layer: com.topjohnwu.magisk/a.b#1
> 04-07 08:21:55.688   511  1190 W SurfaceFlinger: Attempting to set client state on removed layer: Dim Layer for - Task=1002733#0
> 04-07 08:21:55.688   511  1190 W SurfaceFlinger: Attempting to destroy on removed layer: com.topjohnwu.magisk/a.b#1
> 04-07 08:21:55.688   511  1190 W SurfaceFlinger: Attempting to destroy on removed layer: Dim Layer for - Task=1002733#0
> 
> 04-07 08:21:56.209  2487  4540 I ActivityManager: START u0 {dat=file:///storage/emulated/0/Download/F-Droid_Privileged_Extension-2019-03-28v2.zip flg=0x10000000 cmp=com.topjohnwu.magisk/a.f (has extras)} from uid 10306
> 04-07 08:21:56.216   485   485 E ANDR-PERF: IPerf::tryGetService failed!
> 04-07 08:21:56.216   485   485 E QCOM PowerHAL: Failed to acquire lock.
> 04-07 08:21:56.217   485   485 E QCOM PowerHAL: Failed to perform launch boost
> 04-07 08:21:56.236 31365 31365 W ActivityThread: handleWindowVisibility: no activity for token android.os.BinderProxy@cc4cca0
> 04-07 08:21:56.335   485   485 E ANDR-PERF: IPerf::tryGetService failed!
> 04-07 08:21:56.335   485   485 E QCOM PowerHAL: Failed to acquire lock.
> 04-07 08:21:56.338   485   485 E ANDR-PERF: IPerf::tryGetService failed!
> 04-07 08:21:56.338   485   485 E QCOM PowerHAL: Failed to acquire lock.
> 
> 


[magisk-diffs.zip](https://github.com/topjohnwu/Magisk/files/3051149/magisk-diffs.zip)
![Screenshot from 2019-04-07 08-08-32](https://user-images.githubusercontent.com/788766/55677886-dedd4200-590d-11e9-9db8-34c296b915d4.png)

**Comment by topjohnwu** [comment]
Uninstall the old version and reinstall the one from Github

**Comment by m-doescode** [comment]
Note for any future readers: I encountered a similar problem recently because I had actually patched the AP using the f-droid version of Magisk. This caused the update system to fail because it downloads the new version from github, and the f-droid version has a different signature for some reason.

Solution was to stop magisk daemon (`magisk --stop` with su in Termux), uninstall magisk manager, install magisk manager from official github releases, repatch AP and reflash. I was then able to use the update system correctly.
